core/src/main/java/io/jenkins/pluginhealth/scoring/probes/FailingBuildProbe.java

```diff
@@ -0,0 +1,107 @@
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.jenkins.pluginhealth.scoring.probes;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.stream.Stream;
+
+import io.jenkins.pluginhealth.scoring.model.Plugin;
+import io.jenkins.pluginhealth.scoring.model.ProbeResult;
+
+import org.kohsuke.github.GHCheckRun;
+import org.kohsuke.github.GHCommit;
+import org.kohsuke.github.GHRepository;
+import org.kohsuke.github.GitHub;
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Component;
+
+/**
+ * This probe checks whether build failed on Default Branch or not.
+ */
+@Component
+@Order(FailingBuildProbe.ORDER)
+public class FailingBuildProbe extends Probe {
+
+    public static final int ORDER = LastCommitDateProbe.ORDER + 100;
+    public static final String KEY = "failing-buildingProbe";
+
+    @Override
+    protected ProbeResult doApply(Plugin plugin, ProbeContext context) {
+        if (context.getRepositoryName().isPresent()) {
+            return this.success("There is no local repository for plugin " + plugin.getName() + ".");
+        }
+        try {
+            if (repoContainsJenkins(context).toString().equals("Jenkinsfile found")) {
+
+                final GitHub gh = context.getGitHub();
+                final GHRepository repository = gh.getRepository(context.getRepositoryName().orElseThrow());
+                String defaultBranch = repository.getDefaultBranch();
+                GHCommit commit = repository.getCommit(defaultBranch);
+                GHCheckRun checkRun = commit.getCheckRuns().iterator().next();
+                GHCheckRun.Conclusion conclusion = checkRun.getConclusion();
+
+                if (conclusion == GHCheckRun.Conclusion.FAILURE) {
+                    return this.success("Build Failed in Default Branch");
+                } else {
+                    return this.success("Build is Success in Default Branch");
+                }
```

Comment:
Those statements need to be tested.

Reply:
@alecharp can you help me in the testing,  GHCommit ghCommit = new GHCommit(); i cannot set particular commits for testing. any kind of more generic way to test it